raft/src/test/java/org/apache/kafka/raft/KafkaRaftClientTest.java

```diff
@@ -1249,7 +1249,7 @@ public void testChannelWokenUpIfLingerTimeoutReachedDuringAppend(boolean withKip
     @ValueSource(booleans = { true, false })
     public void testHandleEndQuorumRequest(boolean withKip853Rpc) throws Exception {
         int localId = randomReplicaId();
-        int oldLeaderId = randomReplicaId() + 1;
+        int oldLeaderId = localId + 1;
```

Comment:
nice finding. The duplicate id will make only one voter and it become candidate immediately. Hence, the epoch gets increased too.